### ISSUE TITLE
The cost of wieldiong an item from a holster is calculated wrong

### ISSUE THREAD

**Issue by hexagonrecursion** [CleverRaven/Cataclysm-DDA#41835]
# Is your feature request related to a problem? Please describe.

The current formula for the movement point cost of wielding an item from any container you are wearing appears to be:
```
base + wield_container / 2 + 100 * volume / 2L
```
where
- base is "base moves to remove item" (sic) as listed on the pocket info screen
- wield_container is the movement point cost of wielding the container itself
- volume is the volume of the item you are trying to wield

For example it takes 207 (50 + 180/2 + 100 * 1.35L/2L) movement points to wield a loaded SIG 552 (with a 20 round STANAG magaizne) from a survivor harness. This formula is fine for bags and backpacks. In fact I wouldn't bat an eyelid if it was 
```
base + wield_container + 100 * volume / 2L
```
but for holsters, survivor harnesses, belts, shirt pockets etc this formula is completely wrong - it should not include the cost of wielding the holster itself.
```
base + 100 * volume / 2L
```

# Describe the solution you'd like
Introduce a new pocket flag (WIELD_COST_INCLUDES_CONTAINER) to distinguish backpacks from holsters and only add the cost of wielding the container to for items stored in pockets with this flag.

# Describe alternatives you've considered
Somehow make the cost of wielding all shirts, pants, holsters, harnesses, belts etc very low.

# Additional context
Also: the cost of wielding an item is inconsistent with the cost of drawing it from a holster by activating the holster: it takes 530 movement points to draw the SIG 552 from the survivor harness.

**Comment by LcsTen** [comment]
Hello!

I would like to try to take care of this issue.
Firstly, I wanted to know how did you find that formula.
In fact, as #41855 pointed out, there are three formulas, one used while drawing out a weapon by [a]ctivating the holster, another one used by directly [w]earing the weapon, and the last one displayed in the [w]ear menu.

If I understand correctly the game code, there are two "paths" that lead to the ultimate item::on_wield method, which actually allow the character to wield the weapon.

The first one is:
- game::wield (game.cpp:8998)
- avatar::wield (avatar.cpp:1525)
- item::on_wield (item.cpp:4314)

The second one is:
- holster_actor::use (iuse_actor.cpp:2373)
- player::wield_contents (player.cpp:3512)
- item::on_wield (item.cpp:4314)

The first path is probably used by [w]ield and the second one by [a]ctivate.

As expected, the formulas for the two paths are different.
The formulas depend on the "moves", the volume, the type of the weapon, the skill of the character, and the presence of the "NEEDS_UNFOLD" or "SLOW_WIELD" flags.
In your example, we are using a SIG 552, which is a gun who needs unfold because of his folding stock.

For the first path, the resulting formula is:
``max(INVENTORY_HANDLING_PENALTY/2 + min(200,volume/20_ml),0) + max(0,300 - skill*10)``
Where INVENTORY_HANDLING_PENALTY is a constant worth 100.

Assuming we are using a fresh character without any skill in guns, the formula becomes:
``max(50 + min(200,volume/20_ml),0) + 300``

Assuming we are using a skilled character with 10 in gun, the formula becomes:
``max(50 + min(200,volume/20_ml),0) + 200``

For the second path, the resulting formula is:
``max(moves + min(200,volume/20_ml),0)/((skill+10)/10) + max(0,300 - skill*10)``

Assuming we are using a fresh character without any skill in guns, the formula becomes:
``max(moves + min(200,volume/20_ml),0) + 300``

Assuming we are using a skilled character with 10 in gun, the formula becomes:
``max(moves + min(200,volume/20_ml),0)/2 + 200``

There are two differences between these two formulas.

The first one is that the moves in the second formula is replaced by the constant 50 in the first one. moves is the "moves" property in the pocket objects. This means that this attribute is simply ignored. Actually, it doesn't change that much for holsters, as most of them have 50 as moves property value. It does change a lot for backpack though. Most backpack have pretty big moves value, the backpack for example has 300 moves. In-game, it's as if the backpack only has 50 moves. Drawing a gun from a bag is way faster than it should be!

The second one is the way skill is involved in the formula. In both formulas, 10 moves is removed for each skill point, up to a maximum of 300.
There is something more for the second formula though, as the skill is involved in a division.
Let's say both formulas are acting the same way regarding to base_moves_remove and we are removing an heavy weapon (volume/20_ml > 200) from a backpack (300 moves) as a skilled character (LVL 10), the results are:
``(300 + 200) + 200 = 700`` for the first formula,
``(300 + 200)/2 + 200 = 450`` for the second one.
There would be a difference of 250 moves, which is pretty big!

In conclusion, I think that the holster wielding cost is working properly, but a bug makes wielding from a backpack faster. I don't know whether it's intended or not, but it seems like the INVENTORY_HANDLING_PENALTY constant should be replaced by the "moves" property value. The problem is, if it would be indeed replaced, it would almost game-breaking, as wielding from a backpack will be much slower. The skill difference may be intended yet incorrectly implemented. After all, it makes sense that drawing from a holster cost less time when more skilled, while being skilled with gun won't allow you to search your bag for your gun faster. That may be subject for debate, though.

I'm waiting for an approval before making any change.

**Comment by hexagonrecursion** [comment]
[rant]
It has a folding stock?! I did not notice that. CDDA is an incredibly deep and complex game. It is such a shame that its mechanics outpace its UI by so much. If you did not tell me the game modeled folding stocks I would have never found it - right at the bottom of the item description menu it says "integrated mod: folding stock (stock) A folding stock which reduces volume but needs unfolding before use.  Increases the time needed to wield the weapon." It would be a lot easier for new players to make informed choices if the game told them how it calculates all those numbers it shows in its UI. Especially the move cost of the most recent action.
[/rant]

@LcsTen
> Firstly, I wanted to know how did you find that formula.

In chronological order:
1. I noticed that the movement point cost of removing my SIG 552 from my survivor harness **as displayed in the [w]ield menu** was much higher than I expected it to be.
2. I did some code digging formulated a hypothesis that the formula (I'm sorry, but I do not remember where in the code I found it) might be:
```
base + wield_container / 2 + 100 * volume / 2L
```
3. I did a couple of tests and the costs displayed in the menu seemed consistent with this formula
4. Later I noticed that the actual move costs are not consistent with what is shown in the munu and opened #41855

**Comment by LcsTen** [comment]
Ok, I think I see where you found that formula. It is indeed the formula used by the [w]ield menu.
This formula is actually correct for most items, with two exceptions: items hold in holster (where it incorrectly adds the cost of the holster and doesn't perform the skill check) and items with the "NEEDS_UNFOLD" or "SLOW_WIELD" flags (for which there is an pretty big additional penalty lessened with skill).

Ironically, you thought the cost of removing the gun as displayed in the [w]ield menu is high, though the real cost to remove it is about twice higher!

The formula you want is actually already used when drawing a weapon by [a]ctivating the holster.
Did you make tests with [a]ctivate? Does the cost suit you? Remember the SIG 552 has a folding stock, which comes with a big penalty, even with high skill. You should try with weapons which are neither "NEEDS_UNFOLD" nor "SLOW_WIELD".

Another thing I forgot to mention, did you mod the game? I couldn't put the SIG 552 in a survivor harness. If I remember correctly, it may only be put in an XL holster.

There is another difference I forgot between the two formulas: For the first one, obtain_cost_multiplier is taken into account, while it isn't for the second one. This means the cost of drawing a weapon from an holster is not affected by traits such as Packmule. Whether it should or not is then again subject to debate.

**Comment by hexagonrecursion** [comment]
> Another thing I forgot to mention, did you mod the game? I couldn't put the SIG 552 in a survivor harness. If I remember correctly, it may only be put in an XL holster.

Just tested. I can put SIG 552 with a STANAG 20-round magazine into a survivor harness just fine.
```
- OS: Linux
    - OS Version: LSB Version: :core-4.1-amd64:core-4.1-noarch; Distributor ID: Fedora; Description: Fedora release 32 (Thirty Two); Release: 32; Codename: ThirtyTwo; 
- Game Version: 0.E-4395-g83ce24e [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]
```
